storage/util.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+
+	"github.com/coreos/etcd/pkg/fileutil"
+)
+
+func Exist(dir string) bool {
+	return fileutil.Exist(path.Join(dir, defaultFilename))
+}
+
+func SaveTempSnapshot(dir string, index uint64, r io.Reader) error {
```

Comment:
func Save() is good enough. 

if you put this into storage, it should simply read from io.reader and save into the path you give it if the file does not exist before.